paasta_tools/cli/cmds/status.py

```diff
@@ -714,12 +715,27 @@ def status_kubernetes_job_human(
 
 
 def print_flink_status(
-    cluster: str, service: str, instance: str, output: List[str], status, verbose: int
+    cluster: str,
+    service: str,
+    instance: str,
+    output: List[str],
+    flink: Mapping[str, Any],
+    verbose: int,
 ) -> int:
+    status = flink.get("status")
     if status is None:
         output.append(PaastaColors.red("    Flink cluster is not available yet"))
         return 1
 
+    # Since metadata should be available no matter the state, we show it first. If this errors out
+    # then we cannot really do much to recover, because cluster is not in usable state anyway
+    metadata = flink.get("metadata")
+    config_sha = metadata.labels.get("yelp.com/paasta_config_sha")
```

Comment:
I think you can still provide hints here, even if you're going to throw the exception:

```
 config_sha = metadata.labels.get("yelp.com/paasta_config_sha")
if config_sha is None:
    raise ValueError(f"expected config sha on Flink CR. ObjectMeta {metadata}")
if config_sha.startswith("config"):
    config_sha = config_sha[6:]
```